server/api/auth/auth.test.ts

```diff
@@ -0,0 +1,114 @@
+import { randEmail } from '@ngneat/falso';
+import DiscordOauth2 from 'discord-oauth2';
+import { cleanEnv, str } from 'envalid';
+import jwt from 'jsonwebtoken';
+import setCookie, { Cookie } from 'set-cookie-parser';
+
+import * as authLib from 'server/lib/auth';
+import { User } from 'server/models';
+import TestServer from 'server/test/server';
+import { AUTH_COOKIE_NAME } from 'shared/constants';
+
+const env = cleanEnv(process.env, {
+  DISCORD_GUILD_ID: str(),
+});
+
+function mockDiscord(user: Partial<DiscordOauth2.User>, isValid = true ) {
+  return jest.spyOn(authLib, 'getDiscordUserAndGuilds').mockResolvedValueOnce({
+    user: user as DiscordOauth2.User,
+    guilds: [
+      { id: isValid ? env.DISCORD_GUILD_ID : '12345' } as DiscordOauth2.PartialGuild,
+    ],
+  });
+}
```

Comment:
We don't actually want to hit discord oauth API with any requests when we're running tests, so this will mock the response.